wordpress-mobile/WordPress-iOS
# Use MediaProgressCoordinator on hybrid.

This PR does the following:

- Move the MediaProgressCoordinator to a separate file outsize of the AztecPostController
- Refactor the WPPostViewController to use the MediaProgressCoordinator
- Implement some unit tests on MediaProgressCoordinator 

To test:
 - Start the app
 - Make sure the Hybrid editor is active
 - Create a new post
 - Add different media to it
 - Check if progress is reported correctly
 - Check the following scenarios
   - Cancel an upload
   - Fail an upload ( For ex:.disconnect the network in the middle of a upload)
   - Retry a failed upload
   - After a first batch of uploads is done, add another set of media
   - try to switch to html mode or save a post while uploads are still going on
   - try to switch to html mode or save a post while uploads failed 


Needs review: @
